crates/notion-core/src/error/reporter.rs

```diff
@@ -98,8 +95,31 @@ impl ErrorReporter {
     }
 }
 
-fn compose_error_details(err: &NotionError, inner: &Fail) -> String {
-    let mut details = format_error_cause(inner);
+fn compose_error_details(err: &NotionError) -> Option<String> {
+    // Only compose details if there is an underlying cause for the error
+    let mut current = match err.as_fail().cause() {
+        Some(cause) => cause,
+        None => {
+            return None;
+        }
+    };
+    let mut details = String::new();
+
+    // Walk up the tree of causes and include all of them
+    loop {
+        write!(details, "{}", format_error_cause(current))
+            .expect("write! to a String doesn't fail");

```

Comment:
You can avoid the `write!` and `expect()` here by using string concatenation:

```rust
details += &format_error_cause(current);
```

Or you could use `push_str`

```rust
details.push_str(&format_error_cause(current));
```

Reply:
Good call! I was so caught up thinking about formatting that I completely missed that I didn't need the power of the `format!` macros when appending a string value. Updated!